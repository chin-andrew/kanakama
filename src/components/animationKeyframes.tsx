import { keyframes } from 'styled-components';

export const fadeIn = keyframes`
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
`;

export const rotate90 = keyframes`
  0% {
    transform: rotate(0);
  }
  100% {
    transform: rotate(90deg);
  }
`;

export const rotateNeg90 = keyframes`
  0% {
    transform: rotate(90deg);
  }
  100% {
    transform: rotate(0deg);
  }
`;
