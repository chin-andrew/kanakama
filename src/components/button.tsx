import styled from 'styled-components';

export default styled.button`
  align-self: center;
  border: 2px solid black;
  background: none;
  cursor: pointer;
  font-size: 16px;
  letter-spacing: 0.3px;
  margin: 4px;
  outline: none;
  padding: 12px 40px;
  width: 200px;

  :hover {
    background-color: darkgrey;
  }
`;
