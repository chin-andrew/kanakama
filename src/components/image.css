.logo-image {
  height: 75px;
  width: 75px;
}

.question-image {
  max-height: 300px;
  max-width: 300px;
}

@media (max-width: 600px) {
  .question-image {
    max-height: 150px;
    max-width: 150px;
  }
}

.stats-image {
  height: 25px;
  width: 25px;
}
