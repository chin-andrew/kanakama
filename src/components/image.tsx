import React from 'react';

export default (imagePath: string, className: string) => {
  // @ts-ignore
  const reqSvgs = require.context ( '../images', true, /(\.svg)|(\.png)$/ )
  return <img className={className} src={reqSvgs(imagePath)} alt={''} />
}
