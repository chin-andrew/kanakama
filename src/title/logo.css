.logo-container {
  border: 4px solid black;
  margin: 16px;
  width: 150px;
}

.logo-image {
  height: 75px;
  width: 75px;
}