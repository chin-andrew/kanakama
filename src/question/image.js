import React from 'react';
import './image.css'

export default (imagePath, className) => {
  const reqSvgs = require.context ( '../images', true, /(\.svg)|(\.png)$/ )
  return <img className={className} src={reqSvgs(imagePath)} />
}

