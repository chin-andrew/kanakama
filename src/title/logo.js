import React from 'react';
import fetchKanaImage from '../question/image';
import './logo.css';

export default () => {
  return (
    <div className="logo-container">
      {fetchKanaImage('./hiragana-ka.svg', 'logo-image')}
      {fetchKanaImage('./hiragana-na.svg', 'logo-image')}
      {fetchKanaImage('./hiragana-ka.svg', 'logo-image')}
      {fetchKanaImage('./hiragana-ma.svg', 'logo-image')}
    </div>
  )
}