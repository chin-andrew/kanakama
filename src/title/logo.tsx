import React from 'react';
import styled from 'styled-components';
import fetchKanaImage from '../components/image';
import './logo.css';

export const LogoContainer = styled.div`
  border: 4px solid black;
  margin: 16px;
  width: 150px;
`;

export default () => {
  return (
    <LogoContainer>
      {fetchKanaImage('./hiragana-ka.svg', 'logo-image')}
      {fetchKanaImage('./hiragana-na.svg', 'logo-image')}
      {fetchKanaImage('./hiragana-ka.svg', 'logo-image')}
      {fetchKanaImage('./hiragana-ma.svg', 'logo-image')}
    </LogoContainer>
  );
};
