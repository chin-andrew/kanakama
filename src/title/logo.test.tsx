import renderer from 'react-test-renderer';
import sinon from 'sinon';

import * as Image from '../components/image';
import logo from './logo';

describe('logo tests', () => {
  let stub: any;
  beforeEach(() => {
    stub = sinon.stub(Image, 'default');
  });

  afterEach(() => {
    stub.restore();
  });

  it('renders the logo', () => {
    const tree = renderer.create(logo()).toJSON();
    expect(tree).toMatchSnapshot();
    sinon.assert.callCount(stub, 4);
  });
});
