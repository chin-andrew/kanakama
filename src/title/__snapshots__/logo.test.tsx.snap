// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logo tests renders the logo 1`] = `
<div
  className="sc-bdVaJa eARkIM"
/>
`;
